tests/5.1/target_update/test_target_update_iterator.c

```diff
@@ -0,0 +1,78 @@
+//===--- test_target_update_iterator.c --------------------------------------------------------------===//
+//
+//  OpenMP API Version 5.1 Aug 2021
+//
+//  This test usese the target update directive with the iterator clause
+//  to update the iterative values.
+//
+//  This example has been adapted from the 5.2 OpenMP Examples document,
+//  "Multidependences Using Iterators" and "Simple target data and target update
+//  constructs"
+//
+////===--------------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 1024
+
+typedef struct test_struct{
+    size_t len;
+    int *data;
+} test_struct_t;
+
+void init(struct test_struct *s ){
+    s->len = N;
+    for(size_t i = 0; i < s->len; i++){
+        s->data[i] = i;
+    }
+}
+
+void init_again(struct test_struct *s ){
+    s->len = N;
+    for(size_t i = 0; i < s->len; i++){
+        s->data[i] = i+1;
+    }
+}
+
+int test_target_update_iterator() {
+    int errors = 0;
+    int A[N]; // array for testing
+
+    for(int i = 0; i < N; i++){
+        A[i] = 0;
+    }
+    test_struct_t new_struct;
+
+    init(&new_struct);
+    #pragma omp target map(to: new_struct) map(tofrom: A[:N])
```

Comment:
When mapping `new_struct`, its member (`new_struct.data`) should also be explicitly mapped since it is a pointer (compiler cannot know the size of the data referenced by the pointer.)

Reply:
Agree, it needs to be a) mapped and b)attached.

The maps should be:
```
#pragma omp target enter data map(to: new_struct.data[:N])

    #pragma omp target map(to: new_struct, new_struct.data[:N]) map(tofrom: A[:N]) //pointer attachment
    {
        #pragma omp parallel for
```